### PR TITLE
Disable secondary power source by default, fix battery param migration (take 2)

### PR DESCRIPTION
**Pull request by bkueng**
Apparently https://github.com/PX4/Firmware/pull/15616 was still incorrect.

- Fixes param migration, e.g. if BAT_N_CELLS is set, migrates to BAT1_N_CELLS.
- disable secondary module by default. This avoids a GCS showing 2 battery indicators. Alternatively we could also check the 'connected' flag, but this is more explicit.

@DonLakeFlyer does this work for you?

Fixes https://github.com/PX4/Firmware/issues/15929

### CODE REVIEW

File: src/lib/battery/battery.h [hunk 1]
```diff
@@ -193,7 +193,7 @@ class Battery : public ModuleParams
 		}
 	}
 
-	bool isFloatEqual(float a, float b) const { return fabsf(a - b) > FLT_EPSILON; }
+	bool isFloatEqual(float a, float b) const { return fabsf(a - b) < FLT_EPSILON; }
```
Comment (dagar) [inline src/lib/battery/battery.h:196]: We should really do a proper version of this in mathlib with all the special cases handled.

**Review by dagar — APPROVED** [review]
I'd like to get feedback from @DonLakeFlyer and test myself, but otherwise the changes look good.

**Comment by DonLakeFlyer** [comment]
This is the first I'm hearing about a new BAT#_SOURCE parameter. The QGC Power setup currently does have that in the ui. I assume I need to add that. And then after that this looks like exactly what is needed.

But where's the change to not send telemetry for a battery with BAT#_SOURCE=disabled. 

You can verify that QGC will be happy with telemetry by using daily build and you get the correct number of batteries in the toolbar. Also could someone run daily and look at Power setup to see if there is other stuff missing there?

File: src/lib/battery/battery.h (hunk 1 above)
Comment (MaEtUgR) [reply src/lib/battery/battery.h:196]: You mean https://github.com/PX4/Matrix/blob/master/matrix/helper_functions.hpp#L23-L40 ?

**Comment by cuhome** [comment]
@dagar [Can you fix this bug together?](https://github.com/PX4/Firmware/issues/15769)

**Comment by cuhome** [comment]
> @dagar [可以一起解决此错误吗？](https://github.com/PX4/Firmware/issues/15769)

I suggest adding a BAT_can_N_CELLS parameter to fix the uavcan battery bug.

**Comment by bkueng** [comment]
> But where's the change to not send telemetry for a battery with BAT#_SOURCE=disabled.

No further change needed, that's already the effect of `BAT#_SOURCE=disabled`.

> You can verify that QGC will be happy with telemetry by using daily build and you get the correct number of batteries in the toolbar.

I tested it against that. The default will now only show 1 battery, and enabling `BAT2_SOURCE` will show 2.

> Are the voltage/current divider params only valid for some selection of BAT#_SOURCE? Or is that needs for all source types?

Currently only for `Power Module (0)`.

File: src/lib/battery/battery.h (hunk 1 above)
Comment (bkueng) [reply src/lib/battery/battery.h]: Changed to that.

**Review by MaEtUgR — APPROVED** [review]
I wonder how that happened, I remember testing all cases on hardware. Must have been a last minute change after that 😬 
https://github.com/PX4/Firmware/commit/9c4e26c00c593548e24e42a127fdfba9c2a5a996#diff-0eb86e4eec709144f8f3fa953148871cd33a1b6a2332316777c39260e22848a3L191-R195
Thanks for fixing the check @bkueng !

**Comment by DonLakeFlyer** [comment]
Will this get into a point release of stable?

**Pull request merged**